theano/sandbox/gpuarray/opt.py

```diff
@@ -300,6 +300,19 @@ def local_gpualloc_memset_0(node):
             return [new_op(*node.inputs)]
 
 
+# Don't register by default.
+@gof.local_optimizer([GpuAllocEmpty])
+def local_gpua_alloc_empty_to_zeros(node):
+    if isinstance(node.op, GpuAllocEmpty):
+        return [GpuAlloc()(theano.tensor.constant(0, dtype='float32'),
+                           *node.inputs)]
```

Comment:
This will break, you need to call as_gpuarray_varialbe on your 0 value and pass the proper context (which should be node.op.context_name in this case).